// @aws-sdk/client-comprehend/dist-es/commands/BatchDetectSentimentCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { BatchDetectSentimentRequest, BatchDetectSentimentResponse } from "../models/models_0";
import { deserializeAws_json1_1BatchDetectSentimentCommand, serializeAws_json1_1BatchDetectSentimentCommand, } from "../protocols/Aws_json1_1";
var BatchDetectSentimentCommand = (function (_super) {
    __extends(BatchDetectSentimentCommand, _super);
    function BatchDetectSentimentCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    BatchDetectSentimentCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ComprehendClient";
        var commandName = "BatchDetectSentimentCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: BatchDetectSentimentRequest.filterSensitiveLog,
            outputFilterSensitiveLog: BatchDetectSentimentResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    BatchDetectSentimentCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1BatchDetectSentimentCommand(input, context);
    };
    BatchDetectSentimentCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1BatchDetectSentimentCommand(output, context);
    };
    return BatchDetectSentimentCommand;
}($Command));
export { BatchDetectSentimentCommand };
