// @aws-sdk/client-comprehend/dist-es/commands/DescribePiiEntitiesDetectionJobCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribePiiEntitiesDetectionJobRequest, DescribePiiEntitiesDetectionJobResponse } from "../models/models_0";
import { deserializeAws_json1_1DescribePiiEntitiesDetectionJobCommand, serializeAws_json1_1DescribePiiEntitiesDetectionJobCommand, } from "../protocols/Aws_json1_1";
var DescribePiiEntitiesDetectionJobCommand = (function (_super) {
    __extends(DescribePiiEntitiesDetectionJobCommand, _super);
    function DescribePiiEntitiesDetectionJobCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribePiiEntitiesDetectionJobCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ComprehendClient";
        var commandName = "DescribePiiEntitiesDetectionJobCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribePiiEntitiesDetectionJobRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DescribePiiEntitiesDetectionJobResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribePiiEntitiesDetectionJobCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1DescribePiiEntitiesDetectionJobCommand(input, context);
    };
    DescribePiiEntitiesDetectionJobCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1DescribePiiEntitiesDetectionJobCommand(output, context);
    };
    return DescribePiiEntitiesDetectionJobCommand;
}($Command));
export { DescribePiiEntitiesDetectionJobCommand };
