// @aws-sdk/client-comprehend/dist-es/commands/ListKeyPhrasesDetectionJobsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListKeyPhrasesDetectionJobsRequest, ListKeyPhrasesDetectionJobsResponse } from "../models/models_0";
import { deserializeAws_json1_1ListKeyPhrasesDetectionJobsCommand, serializeAws_json1_1ListKeyPhrasesDetectionJobsCommand, } from "../protocols/Aws_json1_1";
var ListKeyPhrasesDetectionJobsCommand = (function (_super) {
    __extends(ListKeyPhrasesDetectionJobsCommand, _super);
    function ListKeyPhrasesDetectionJobsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListKeyPhrasesDetectionJobsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ComprehendClient";
        var commandName = "ListKeyPhrasesDetectionJobsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListKeyPhrasesDetectionJobsRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListKeyPhrasesDetectionJobsResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListKeyPhrasesDetectionJobsCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1ListKeyPhrasesDetectionJobsCommand(input, context);
    };
    ListKeyPhrasesDetectionJobsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1ListKeyPhrasesDetectionJobsCommand(output, context);
    };
    return ListKeyPhrasesDetectionJobsCommand;
}($Command));
export { ListKeyPhrasesDetectionJobsCommand };
