// @aws-sdk/client-comprehend/dist-es/commands/ListDocumentClassifierSummariesCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListDocumentClassifierSummariesRequest, ListDocumentClassifierSummariesResponse } from "../models/models_0";
import { deserializeAws_json1_1ListDocumentClassifierSummariesCommand, serializeAws_json1_1ListDocumentClassifierSummariesCommand, } from "../protocols/Aws_json1_1";
var ListDocumentClassifierSummariesCommand = (function (_super) {
    __extends(ListDocumentClassifierSummariesCommand, _super);
    function ListDocumentClassifierSummariesCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListDocumentClassifierSummariesCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ComprehendClient";
        var commandName = "ListDocumentClassifierSummariesCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListDocumentClassifierSummariesRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListDocumentClassifierSummariesResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListDocumentClassifierSummariesCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1ListDocumentClassifierSummariesCommand(input, context);
    };
    ListDocumentClassifierSummariesCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1ListDocumentClassifierSummariesCommand(output, context);
    };
    return ListDocumentClassifierSummariesCommand;
}($Command));
export { ListDocumentClassifierSummariesCommand };
