// @aws-sdk/client-comprehend/dist-es/commands/ListPiiEntitiesDetectionJobsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListPiiEntitiesDetectionJobsRequest, ListPiiEntitiesDetectionJobsResponse } from "../models/models_0";
import { deserializeAws_json1_1ListPiiEntitiesDetectionJobsCommand, serializeAws_json1_1ListPiiEntitiesDetectionJobsCommand, } from "../protocols/Aws_json1_1";
var ListPiiEntitiesDetectionJobsCommand = (function (_super) {
    __extends(ListPiiEntitiesDetectionJobsCommand, _super);
    function ListPiiEntitiesDetectionJobsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListPiiEntitiesDetectionJobsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ComprehendClient";
        var commandName = "ListPiiEntitiesDetectionJobsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListPiiEntitiesDetectionJobsRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListPiiEntitiesDetectionJobsResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListPiiEntitiesDetectionJobsCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1ListPiiEntitiesDetectionJobsCommand(input, context);
    };
    ListPiiEntitiesDetectionJobsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1ListPiiEntitiesDetectionJobsCommand(output, context);
    };
    return ListPiiEntitiesDetectionJobsCommand;
}($Command));
export { ListPiiEntitiesDetectionJobsCommand };
