// @aws-sdk/client-comprehend/dist-es/commands/StopDominantLanguageDetectionJobCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { StopDominantLanguageDetectionJobRequest, StopDominantLanguageDetectionJobResponse } from "../models/models_0";
import { deserializeAws_json1_1StopDominantLanguageDetectionJobCommand, serializeAws_json1_1StopDominantLanguageDetectionJobCommand, } from "../protocols/Aws_json1_1";
var StopDominantLanguageDetectionJobCommand = (function (_super) {
    __extends(StopDominantLanguageDetectionJobCommand, _super);
    function StopDominantLanguageDetectionJobCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    StopDominantLanguageDetectionJobCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ComprehendClient";
        var commandName = "StopDominantLanguageDetectionJobCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: StopDominantLanguageDetectionJobRequest.filterSensitiveLog,
            outputFilterSensitiveLog: StopDominantLanguageDetectionJobResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    StopDominantLanguageDetectionJobCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1StopDominantLanguageDetectionJobCommand(input, context);
    };
    StopDominantLanguageDetectionJobCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1StopDominantLanguageDetectionJobCommand(output, context);
    };
    return StopDominantLanguageDetectionJobCommand;
}($Command));
export { StopDominantLanguageDetectionJobCommand };
