// @aws-sdk/client-translate/dist-es/commands/CreateParallelDataCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateParallelDataRequest, CreateParallelDataResponse } from "../models/models_0";
import { deserializeAws_json1_1CreateParallelDataCommand, serializeAws_json1_1CreateParallelDataCommand, } from "../protocols/Aws_json1_1";
var CreateParallelDataCommand = (function (_super) {
    __extends(CreateParallelDataCommand, _super);
    function CreateParallelDataCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateParallelDataCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "TranslateClient";
        var commandName = "CreateParallelDataCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateParallelDataRequest.filterSensitiveLog,
            outputFilterSensitiveLog: CreateParallelDataResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateParallelDataCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1CreateParallelDataCommand(input, context);
    };
    CreateParallelDataCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1CreateParallelDataCommand(output, context);
    };
    return CreateParallelDataCommand;
}($Command));
export { CreateParallelDataCommand };
